Content.Server/Flash/FlashSystem.cs

```diff
@@ -25,6 +26,15 @@ public override void Initialize()
             SubscribeLocalEvent<FlashComponent, UseInHandEvent>(OnUseInHand);
 
             SubscribeLocalEvent<FlashComponent, ExaminedEvent>(OnExamined);
+            SubscribeLocalEvent<FlashAreaOnCollide, StartCollideEvent>(HandleCollide);
+        }
+
+        private void HandleCollide(EntityUid uid, FlashAreaOnCollide component, StartCollideEvent args)
+        {
+            if (component.Flashed) return;
+
+            FlashableComponent.FlashAreaHelper(component.Owner, component.Range, component.Duration);
```

Comment:
a future idea for ECS flash would be to make this helper a public entity system method